modules/nixos/services/bobaboard/default.nix

```diff
@@ -1,16 +1,92 @@
-{ lib, config, ... }:
+{ lib, config, inputs, system, pkgs, ... }:
 
 let
   cfg = config.services.bobaboard;
+  bobabackend-packages = inputs.boba-backend.packages."${system}";
 
   inherit (lib) mkEnableOption mkIf;
 in
 {
   options.services.bobaboard = {
     enable = mkEnableOption "BobaBoard";
   };
-
   config = mkIf cfg.enable {
     # @TODO(jakehamilton): Add configuration for BobaBoard here.
+
+    networking.firewall.allowedTCPPorts = [
+      4200
+    ];
+    
+    users = {
+      users.bobaboard = {
+        isSystemUser = true;
+        group = "bobaboard";
+      };
+
+      groups.bobaboard = {};
+    };
+
+    services.postgresql = {
+      enable = true;
+      package = pkgs.postgresql_12;
```

Comment:
This should be configurable with an option.

Reply:
The SQL version? Shouldn't it be up to the service to declare which SQL version it needs?

Reply:
Typically we set a default and allow it to be overridden. You never know when you'll need to swap out the package!